### PR TITLE
Added support to preserve the order when converting XML to JSON

### PR DESCRIPTION
**Pull request by susinda**
Introduced (overload)method XML.toJSONObject(String string, boolean preserveOrder)

### CODE REVIEW

**Comment by stleary** [comment]
Thanks, but can't accept this pull request. The spec says JSON objects are unordered.

**Comment by susinda** [comment]
Hi Stleary
Thanks for looking my PR.
What I meant by order is to maintain the same order as in xml (when converting xml to json). For simple xml files order is not changed (even without this fix) issue comes for complex xmls. If we maintain the same order, is it a violation of spec.

Thanks
Susinda

**Comment by stleary** [comment]
Maintaining order in JSON objects, even for a good reason, would cause JSON-Java to diverge from the spec, and lesson its value as a reference app. If you check the closed pull requests, this is probably the 2nd most popular topic, after Mavenizing the project.

**Comment by johnjaylward** [comment]
By definition, if you are converting to JSON, it means you don't care what order your object keys are in. If you need ordered keys, you should use a different data interchange format.

**Comment by susinda** [comment]
Thanks stleary and johnjaylward.
Understood and agree with your explanation.  Is there any chance that you could make this functionality extendable so that ppl(like me) who needs the order can extend and use it.
Anyway i'd like to thank you and all the developers for your effort on development and maintenance this.

File: XML.java [hunk 1]
```diff
@@ -332,6 +332,37 @@ public static JSONObject toJSONObject(String string) throws JSONException {
...
+     * @param boolean 
+     *            flag saying whether to maintain order or not.
+     * @return A JSONObject containing the structured data from the XML string.
+     * @throws JSONException
+     */
+    public static JSONObject toJSONObject(String string, boolean preserveOrder) throws JSONException {
+        if (!preserveOrder) {
+        	return toJSONObject(string);
+        } else {
+        	JSONObject jo = new OrderedJSONObject();
```
Comment (meetjoeblack) [inline XML.java]: As I see we have new JSONObject() creation inside XML#parse https://github.com/stleary/JSON-java/blob/28e09dc4938e65bbf022302972559ac0179b5ebc/XML.java#L327 method and your changes doesn't cover this case.

**Pull request closed without merging**